app/graphql/types/profile.rb

```diff
@@ -70,6 +71,12 @@ class Profile < Types::BaseObject
 
     field :major_os_version, String, null: false
 
+    def ssg_version
+      ::RecordLoader.for(::Xccdf::Benchmark)
+                    .load(object.benchmark_id)
+                    .then(&:version)
+    end
```

Comment:
Nice! This will also be needed in the REST API - perhaps we could put it in the Profile model itself?

Reply:
The attribute on the REST will be added via  RHICOMPL-1034.

An attribute on the Profile model would make sense. Do you know how to pre-load benchmark for all profiles? Should we do a default scope that would `include(:benchmark)`?

Reply:
It may make sense to use active record directly with something like:

```rb
belongs_to :ssg_version, through: :benchmark
```

idk if that actually works, but it should automatically include the associated benchmark. Otherwise, like you say, we'll need to `include(:benchmark)`.

Reply:
I guess that would only work if the `:ssg_version` was an association to some other model.

I had in mind this:
```
class Profile
  default_scope { includes(:benchmark) }
end
```

Reply:
`default_scope` is a bit... scary tbh. I'd rather avoid it and instead include benchmark manually in the controller layer or create a named scope to include the benchmark where it's actually needed (i.e. `with_benchmarks` or similar).